pkgsMusl.gdb does not compile

**To Reproduce**

`nix-build -A pkgsMusl.gdb`

```
...
  CXX    gdbsupport/format-ipa.o
  CXX    gdbsupport/print-utils-ipa.o
  CXX    gdbsupport/rsp-low-ipa.o
  CXX    gdbsupport/safe-strerror-ipa.o
  CXX    gdbsupport/tdesc-ipa.o
  CXX    linux-amd64-ipa.o
  CXX    linux-x86-tdesc-ipa.o
  CXX    regcache-ipa.o
  CXX    remote-utils-ipa.o
  CXX    tdesc-ipa.o
  CXX    tracepoint-ipa.o
  CXX    utils-ipa.o
  CXXLD  libinproctrace.so
/nix/store/56jgzxaqbdm57jaa7719klj38x487ja1-binutils-2.31.1/bin/ld: ax-ipa.o: in function `ax_printf(unsigned long, unsigned long, char const*, int, unsigned long*) [clone .isra.0]':
/build/gdb-10.1/_build/gdbserver/../../gdbserver/ax.cc:928: undefined reference to `libintl_gettext'
/nix/store/56jgzxaqbdm57jaa7719klj38x487ja1-binutils-2.31.1/bin/ld: /build/gdb-10.1/_build/gdbserver/../../gdbserver/ax.cc:837: undefined reference to `libintl_gettext'
/nix/store/56jgzxaqbdm57jaa7719klj38x487ja1-binutils-2.31.1/bin/ld: /build/gdb-10.1/_build/gdbserver/../../gdbserver/ax.cc:889: undefined reference to `libintl_gettext'
/nix/store/56jgzxaqbdm57jaa7719klj38x487ja1-binutils-2.31.1/bin/ld: ax-ipa.o: in function `gdb_eval_agent_expr(eval_agent_expr_context*, agent_expr*, unsigned long*)':
/build/gdb-10.1/_build/gdbserver/../../gdbserver/ax.cc:1334: undefined reference to `libintl_gettext'
/nix/store/56jgzxaqbdm57jaa7719klj38x487ja1-binutils-2.31.1/bin/ld: gdbsupport/common-utils-ipa.o: in function `xstrvprintf(char const*, __va_list_tag*)':
/build/gdb-10.1/_build/gdbserver/../../gdbserver/../gdbsupport/common-utils.cc:57: undefined reference to `libintl_gettext'
/nix/store/56jgzxaqbdm57jaa7719klj38x487ja1-binutils-2.31.1/bin/ld: gdbsupport/common-utils-ipa.o:/build/gdb-10.1/_build/gdbserver/../../gdbserver/../gdbsupport/common-utils.cc:69: more undefined references to `libintl_gettext' follow
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:389: libinproctrace.so] Error 1
make[2]: Leaving directory '/build/gdb-10.1/_build/gdbserver'
make[1]: *** [Makefile:9365: all-gdbserver] Error 2
make[1]: Leaving directory '/build/gdb-10.1/_build'
make: *** [Makefile:854: all] Error 2
builder for '/nix/store/vqjcmvp3gcjr86ylqbmqn9x146qpf4yn-gdb-10.1.drv' failed with exit code 2
```

6d2cad8a4ed76e42407546bc6b4c717fd9e1a535 (updating gdb, nothing else) is to blame

**Notify maintainers**

@pierron @globin @lsix 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - pkgsMusl.gdb
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/tools/misc/gdb/default.nix
+++ b/pkgs/development/tools/misc/gdb/default.nix
@@ -82,7 +82,8 @@ stdenv.mkDerivation rec {
     "--with-mpfr=${mpfr.dev}"
     "--with-expat" "--with-libexpat-prefix=${expat.dev}"
     "--with-auto-load-safe-path=${builtins.concatStringsSep ":" safePaths}"
-  ] ++ lib.optional (!pythonSupport) "--without-python";
+  ] ++ lib.optional (!pythonSupport) "--without-python"
+    ++ lib.optional stdenv.hostPlatform.isMusl "--disable-nls";
 
   postInstall =
     '' # Remove Info files already provided by Binutils and other packages.

